src/main/java/org/assertj/core/error/ShouldNotStartWith.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ * <p/>
+ * Copyright 2012-2015 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import org.assertj.core.internal.ComparisonStrategy;
+import org.assertj.core.internal.StandardComparisonStrategy;
+
+/**
+ * Creates an error message indicating that an assertion that verifies that a group of elements does not start with a
+ * given value or sequence of values failed. A group of elements can be a collection, an array or a {@code String}.
+ *
+ * @author Michal Kordas
+ */
+public class ShouldNotStartWith extends BasicErrorMessageFactory {
+
+  /**
+   * Creates a new <code>{@link ShouldNotStartWith}</code>.
+   *
+   * @param actual the actual value in the failed assertion.
+   * @param expected the value or sequence of values that {@code actual} is expected to not start with.
+   * @param comparisonStrategy the {@link ComparisonStrategy} used to evaluate assertion.
+   * @return the created {@code ErrorMessageFactory}.
+   */
+  public static ErrorMessageFactory shouldNotStartWith(Object actual, Object expected,
+      ComparisonStrategy comparisonStrategy) {
+    return new ShouldNotStartWith(actual, expected, comparisonStrategy);
+  }
+
+  /**
+   * Creates a new <code>{@link ShouldNotStartWith}</code>.
+   *
+   * @param actual the actual value in the failed assertion.
+   * @param expected the value or sequence of values that {@code actual} is expected to not start with.
+   * @return the created {@code ErrorMessageFactory}.
+   */
+  public static ErrorMessageFactory shouldNotStartWith(Object actual, Object expected) {
+    return new ShouldNotStartWith(actual, expected, StandardComparisonStrategy.instance());
+  }
+
+  private ShouldNotStartWith(Object actual, Object expected, ComparisonStrategy comparisonStrategy) {
+    super("%nExpecting:%n <%s>%nto not start with:%n <%s>%n%s", actual, expected, comparisonStrategy);
```

Comment:
Sorry this is confusing ...

I still see "to not start" instead of "not to start"
> +    super("%nExpecting:%n <%s>%nto not start with:%n <%s>%n%s", actual,
expected, comparisonStrategy);

where I would like
super("%nExpecting:%n  <%s>%n*not to start with*:%n  <%s>%n%s", actual,
expected, comparisonStrategy);

Anyway if you don't see what I mean, I'll just fix that when integrating
your PR, enough time wasted.


On Tue, Apr 21, 2015 at 9:37 AM, Michał Kordas <notifications@github.com>
wrote:

> In src/main/java/org/assertj/core/error/ShouldNotStartWith.java
> <https://github.com/joel-costigliola/assertj-core/pull/378#discussion_r28732020>
> :
>
> > +    return new ShouldNotStartWith(actual, expected, comparisonStrategy);
> > +  }
> > +
> > +  /**
> > +   * Creates a new <code>{@link ShouldNotStartWith}</code>.
> > +   *
> > +   * @param actual the actual value in the failed assertion.
> > +   * @param expected the value or sequence of values that {@code actual} is expected to not start with.
> > +   * @return the created {@code ErrorMessageFactory}.
> > +   */
> > +  public static ErrorMessageFactory shouldNotStartWith(Object actual, Object expected) {
> > +    return new ShouldNotStartWith(actual, expected, StandardComparisonStrategy.instance());
> > +  }
> > +
> > +  private ShouldNotStartWith(Object actual, Object expected, ComparisonStrategy comparisonStrategy) {
> > +    super("%nExpecting:%n <%s>%nto not start with:%n <%s>%n%s", actual, expected, comparisonStrategy);
>
>
>    1. Can you tell me where exactly indentation is wrong? super has 4
>    spaces, because it's inside method.
>    2. to not start replaced with not to start.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joel-costigliola/assertj-core/pull/378/files#r28732020>
> .
>